File: TraningAPI/View/LicenceListView/LicenceListViewController.m
//
//  LicenceListViewController.m
//  TraningAPI
//
//  Created by 高橋慧 on 2021/06/23.
//  Copyright © 2021 k-takahashi. All rights reserved.
//

#import "LicenceListViewController.h"

@interface LicenceListViewController ()

@end

@implementation LicenceListViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.licenceTableView = [[UITableView alloc] initWithFrame:CGRectMake(0, 0, [[UIScreen mainScreen] bounds].size.width, [[UIScreen mainScreen] bounds].size.height) style:UITableViewStyleGrouped];
    self.licenceList = Common.libraryTitle;
    
    self.view.backgroundColor = UIColor.systemGray6Color;
    self.navigationItem.title = Common.licenceTitle;
    
    [self tableViewConfigure];
    [self.view addSubview:self.licenceTableView];
    [self tableViewLayout];
}
- (void)tableViewConfigure {
    self.licenceTableView.delegate = self;
    self.licenceTableView.dataSource = self;
    [self.licenceTableView registerNib:[UINib nibWithNibName: LicenceTableViewCell.cellIdentifer bundle:nil] forCellReuseIdentifier: LicenceTableViewCell.cellIdentifer];
    self.licenceTableView.backgroundColor = UIColor.whiteColor;
}
- (void)tableViewLayout {
    self.licenceTableView.translatesAutoresizingMaskIntoConstraints = false;
    [self.licenceTableView.topAnchor constraintEqualToAnchor: self.view.topAnchor].active = true;
    [self.licenceTableView.widthAnchor constraintEqualToAnchor:self.view.widthAnchor].active = true;
    [self.licenceTableView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor].active = true;
}
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    LicenceContentViewController *nextVC = [StoryBoard performLicenceContentView:self licencesID:indexPath.row];
    [StoryBoard push:self toViewController:nextVC];
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 60;
}
- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.licenceList.count;
}
- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
    LicenceTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:LicenceTableViewCell.cellIdentifer forIndexPath:indexPath];
    [cell licenceNameConfigure: self.licenceList[indexPath.row]];
    return cell;
}



@end

File: TraningAPI/View/LicenceListView/LicenceListViewController.swift
//
//  LicenceListViewController.swift
//  TraningAPI
//
//  Created by 高橋慧 on 2021/04/28.
//  Copyright © 2021 k-takahashi. All rights reserved.
//

import UIKit

class LicenceListViewController: UIViewController {
    // ライセンスリスト
    private let licenceTableView:UITableView = UITableView(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height), style: .grouped)
    private let licenceList = ConstEntity.LIBRARYTITLE
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .systemGray6
        self.navigationController?.navigationBar.isHidden = false
        self.navigationItem.title = ConstEntity.LICENCETITLE
        
        // メニューリスト設定
        tableViewConfigure()
        self.view.addSubview(licenceTableView)
        // メニューリストレイアウト指定
        tableViewLayout()
        // Do any additional setup after loading the view.
    }
    /// メニューリスト設定
       private func tableViewConfigure(){
           licenceTableView.delegate = self
           licenceTableView.dataSource = self
           licenceTableView.translatesAutoresizingMaskIntoConstraints = false
           licenceTableView.register(UINib(nibName: LicenceTableViewCell.className, bundle: nil), forCellReuseIdentifier: LicenceTableViewCell.className)
           licenceTableView.backgroundColor = .white
       }
       ///メニューリスト レイアウト指定
       private func tableViewLayout(){
           licenceTableView.translatesAutoresizingMaskIntoConstraints = false
           licenceTableView.topAnchor.constraint(equalTo: self.view.topAnchor).isActive = true
           licenceTableView.widthAnchor.constraint(equalTo: self.view.widthAnchor).isActive = true
           licenceTableView.bottomAnchor.constraint(equalTo: self.view.bottomAnchor).isActive = true
       }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
extension LicenceListViewController: UITableViewDelegate{
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 60
    }
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        StoryBoard.performNavi(id: indexPath.row, to: Board.licenceContent, from: self)
    }
}
extension LicenceListViewController: UITableViewDataSource{
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return licenceList.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        guard let cell = tableView.dequeueReusableCell(withIdentifier: LicenceTableViewCell.className, for: indexPath) as? LicenceTableViewCell else { return UITableViewCell() }
        cell.licenceNameConfigure(text: licenceList[indexPath.row])
        return cell
    }
}
